lab-cameron/lib/fs-readfile.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+const fs = require('fs');
+
+// I LEFT IN THESE CONSOLE LOGS TO GIVE MY TESTS MORE CONTEXT AND AS PROOF OF ASYNCHRONICITY //
+
+module.exports = function(paths) {
+  let buffers = [];
+  fs.readFile(paths[0], function(err, data) {
+    if (err) throw err;
+    console.log('utf-8 format: ', data.toString('utf-8', 0, 8));
+    buffers.push(data.toString('hex', 0, 8));
+
+    fs.readFile(paths[1], function(err, data) {
+      if (err) throw err;
+      console.log('utf-8 format: ', data.toString('utf-8', 0, 8));
+      buffers.push(data.toString('hex', 0, 8));
+
+      fs.readFile(paths[2],function(err, data) {
+        if (err) throw err;
+        console.log('utf-8 format: ', data.toString('utf-8', 0, 8));
+        buffers.push(data.toString('hex', 0, 8));
+      });
+    });
+  });
+
+  return buffers;
```

Comment:
``return`` actually doesn't work in async Node methods. To get access to data in other modules, you'll have to use a callback that is at the end of your third ``fs.readFile`` method. Here's how I rewrote your code:

```
'use strict';

const fs = require('fs');

// I LEFT IN THESE CONSOLE LOGS TO GIVE MY TESTS MORE CONTEXT AND AS PROOF OF ASYNCHRONICITY //

module.exports = function(paths, callback) {
  let buffers = [];
  fs.readFile(paths[0], function(err, data) {
    if (err) throw err;
    buffers.push(data.toString('hex', 0, 8));

    fs.readFile(paths[1], function(err, data) {
      if (err) throw err;
      buffers.push(data.toString('hex', 0, 8));

      fs.readFile(paths[2],function(err, data) {
        if (err) throw err;

        buffers.push(data.toString('hex', 0, 8));
        if (callback) return callback(buffers);
      });
    });
  });
};
```